packages/react/src/components/dateInput/components/monthTable/MonthTable.tsx

```diff
@@ -15,10 +17,17 @@ export type MonthTableProps = {
 
 export const MonthTable = (props: MonthTableProps) => {
   const { locale } = useContext(DatePickerContext);
-  const { month } = props;
-
-  const weeksInterval = { start: startOfMonth(month), end: endOfMonth(month) };
-  const weeks = eachWeekOfInterval(weeksInterval, { locale });
+  const { month }: { month: Date } = props;
+  const start: Date = startOfMonth(month);
+  const end: Date = endOfMonth(month);
+  const monthCalendarWeeks: number = getWeeksInMonth(month, { weekStartsOn: 1 });
+  const maxAmountOfCalendarWeeks = 6;
+  const weeksEnd: Date =
+    monthCalendarWeeks === maxAmountOfCalendarWeeks
+      ? end
+      : add(end, { weeks: maxAmountOfCalendarWeeks - monthCalendarWeeks }); // To ensure equal height month tables, we need to add extra week rows at the end of the table.
+  const weeksInterval: { start: Date; end: Date } = { start, end: weeksEnd };
+  const weeks: Date[] = eachWeekOfInterval(weeksInterval, { locale });
```

Comment:
Wonder if it would be a good idea to extract the process of determining the weeks into its own function? The variables above the line `const weeks: Date[]...` all seem to relate to determining the weeks? Then again, the user would then have to jump more while reading the code, but perhaps the variables above would be more clearly scoped with a meaning 🤔